Solutions/Corvus.Leasing.Abstractions/Corvus/Leasing/LeasePolicy.cs

```diff
@@ -11,12 +11,19 @@ namespace Corvus.Leasing
     /// </summary>
     public class LeasePolicy
     {
-        private string name;
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LeasePolicy"/> class.
+        /// </summary>
+        /// <param name="name">The <see cref="Name"/>. If null, a generated value will be used.</param>
+        public LeasePolicy(string? name)
```

Comment:
By adding this constructor you've removed the default constructor the compiler was generating. So now, I have to construct this by passing a `name` argument, making that `null` if I want a default name.

The loss of a default ctor is a breaking change. Admittedly, this looked broken before, in that the logic for setting a default name from a `Guid` didn't seem to make a lot of sense before. (It looks like it only kicked in if you explicitly set `Name` to `null`, which is even weirder.)

Is there a reason not to just have a zero-args constructor for when you don't want to specify a name, and only to use this one when you want to bring your own name? (And then this would become `string`, not `string?`)

Reply:
Agreed - I know what I was trying to achieve with this, but the outcome is not good!